src/debug.c

```diff
@@ -1952,7 +1952,7 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
         serverLog(LL_WARNING,
         "Accessing address: %p", (void*)info->si_addr);
     }
-    if (info->si_code <= SI_USER && info->si_pid != -1) {
+    if (info->si_code == SI_USER && info->si_pid != -1) {
```

Comment:
If `info->si_code` equals `SI_USER`, is it safe to say that si_pid must be valid value (i.e. not eq -1)?

Reply:
not sure what's safe and what's not.. too many systems.
i could have always printed the raw data (si_code, si_pid, etc), but preferred not to mislead the reader.
my original attempt was just to match `pid` to -1, but then noticed that on some systems it's not reset.
you may be right that there's no longer need for that check, but on the other hand, i suppose it can't do any damage.

Reply:
If don't have an easy answer, then agree to leave it as is. It wouldn't do any harm anyway.